File: Components/Reporting/VB/ASPxReportDemos/App_Code/DataSourceWizardConnectionStringsProvider.vb
Imports Microsoft.VisualBasic
Imports System.Linq
Imports System.Configuration
Imports System.Collections.Generic
Imports DevExpress.DataAccess.Web
Imports DevExpress.DataAccess.ConnectionParameters

Public Class DataSourceWizardConnectionStringsProvider
	Implements IDataSourceWizardConnectionStringsProvider
	Private connectionStrings As Dictionary(Of String, String)
	Public Sub New()
		connectionStrings = New Dictionary(Of String, String)()
		connectionStrings.Add("Northwind", ConfigurationManager.ConnectionStrings("NWindConnectionString").ConnectionString)
		connectionStrings.Add("Cars", ConfigurationManager.ConnectionStrings("CarsDBConnectionString").ConnectionString)
		connectionStrings.Add("Contacts", ConfigurationManager.ConnectionStrings("ContactsConnectionString").ConnectionString)
		connectionStrings.Add("Countries", ConfigurationManager.ConnectionStrings("CountriesDBConnectionString").ConnectionString)
	End Sub
	Private Function GetConnectionDescriptions() As Dictionary(Of String, String) Implements IDataSourceWizardConnectionStringsProvider.GetConnectionDescriptions
		Return connectionStrings.ToDictionary(Function(k) k.Key, Function(k) k.Key)
	End Function
	Private Function GetDataConnectionParameters(ByVal name As String) As DataConnectionParametersBase Implements IDataSourceWizardConnectionStringsProvider.GetDataConnectionParameters
		Return New CustomStringConnectionParameters(connectionStrings(name))
	End Function
End Class

File: Components/Reporting/CS/MVCxReportDemos/Code/DataSourceWizardConnectionStringsProvider.cs
using System.Linq;
using System.Configuration;
using System.Collections.Generic;
using DevExpress.DataAccess.Web;
using DevExpress.DataAccess.ConnectionParameters;

public class DataSourceWizardConnectionStringsProvider : IDataSourceWizardConnectionStringsProvider {
    Dictionary<string, string> connectionStrings;
    public DataSourceWizardConnectionStringsProvider() {
        connectionStrings = new Dictionary<string, string>();
        connectionStrings.Add("Northwind", ConfigurationManager.ConnectionStrings["NWindConnectionString"].ConnectionString);
        connectionStrings.Add("Cars", ConfigurationManager.ConnectionStrings["CarsDBConnectionString"].ConnectionString);
        connectionStrings.Add("Contacts", ConfigurationManager.ConnectionStrings["ContactsConnectionString"].ConnectionString);
        connectionStrings.Add("Countries", ConfigurationManager.ConnectionStrings["CountriesDBConnectionString"].ConnectionString);
    }
    Dictionary<string, string> IDataSourceWizardConnectionStringsProvider.GetConnectionDescriptions() {
        return connectionStrings.ToDictionary(k => k.Key, k => k.Key);
    }
    DataConnectionParametersBase IDataSourceWizardConnectionStringsProvider.GetDataConnectionParameters(string name) {
        return new CustomStringConnectionParameters(connectionStrings[name]);
    }
}
